core/src/utils/test/playwright/generator.ts

```diff
@@ -38,61 +46,89 @@ interface TestConfigOption {
+       * compatibility, we will not include the theme in the test
+       * title if the theme is set to light.
+       */
+      return `${title} - ${theme}/${direction}`;
+    }
+
+    return `${title} - ${theme}/${direction}/${themeMode}`;
   }
 
-  return `${title} - ${mode}/${direction}/${theme}`;
+  return `${title} - ${theme}/${mode}/${direction}/${themeMode}`;
 };
 
 /**
  * Generates a unique filename based on a base filename
  * and a test config.
  */
 const generateScreenshotName = (fileName: string, config: TestConfig): string => {
-  const { mode, direction, theme } = config;
+  const { theme, direction, themeMode, mode } = config;
 
-  if (theme === 'light') {
+  if (theme === mode) {
     /**
-     * Ionic has many existing tests that existed prior to
-     * the introduction of theme testing. To maintain backwards
-     * compatibility, we will not include the theme in the screenshot
-     * name if the theme is set to light.
+     * Fallback to the old screenshot naming convention
+     * when the theme and mode are the same.
      */
-    return `${fileName}-${mode}-${direction}.png`;
+    if (themeMode === 'light') {
+      /**
+       * Ionic has many existing tests that existed prior to
+       * the introduction of theme testing. To maintain backwards
+       * compatibility, we will not include the theme in the screenshot
+       * name if the theme is set to light.
+       */
+      return `${fileName}-${theme}-${direction}.png`;
+    }
+
+    return `${fileName}-${theme}-${direction}-${themeMode}.png`;
   }
 
-  return `${fileName}-${mode}-${direction}-${theme}.png`;
+  return `${fileName}-${theme}-${mode}-${direction}-${themeMode}.png`;
 };
 
 /**
  * Given a config generate an array of test variants.
  */
 export const configs = (testConfig: TestConfigOption = DEFAULT_TEST_CONFIG_OPTION): TestUtilities[] => {
-  const { modes, directions } = testConfig;
+  const { modes, themes, directions } = testConfig;
 
   const configs: TestConfig[] = [];
 
   /**
    * If certain options are not provided,
    * fall back to the defaults.
    */
-  const processedMode = modes ?? DEFAULT_MODES;
+  const processedModes = modes ?? DEFAULT_MODES;
+  const processedTheme = themes ?? DEFAULT_THEMES;
   const processedDirection = directions ?? DEFAULT_DIRECTIONS;
-  const processedTheme = testConfig.themes ?? DEFAULT_THEMES;
-
-  processedMode.forEach((mode) => {
-    processedDirection.forEach((direction) => {
-      processedTheme.forEach((theme) => {
-        configs.push({ mode, direction, theme });
+  const processedThemeMode = testConfig.themeModes ?? DEFAULT_THEME_MODES;
+
+  processedModes.forEach((mode) => {
+    processedTheme.forEach((theme) => {
+      if (mode === 'md' && theme === 'ios') {
+        // Prevent using Android platform with iOS theme
+        return;
+      }
+      if (mode === 'ios' && theme === 'md') {
+        // Prevent using iOS platform with Material Design theme
+        return;
+      }
```

Comment:
Would it be possible to use the new `isModeValidForTheme` util in `ionic-global.ts` instead, to make this code easier to maintain?

Reply:
I can look further into this. Originally we avoided coupling any source code to the test infrastructure to avoid bundling the test infra in the published package.